### PR TITLE
[FEATURE] New algorithm for creating vector layer from raster layer's extent

### PR DESCRIPTION
**Pull request by nyalldawson**
Allows creation of a new vector layer with a single feature containing a raster layer's extent. Previously this was only possible for vector layers.

### CODE REVIEW

**Comment by nirvn** [comment]
@nyalldawson , any reason for having two separate algorithms here?

**Comment by nyalldawson** [comment]
> any reason for having two separate algorithms here?

I internally debated this a bit. Currently the other alg also has an option to create bounding boxes for features, which makes no sense for rasters. BUT I want to drop that option anyway (since there's a dedicated bounding box alg for this purpose)

If we had one unified algorithm we'd also lose:
- the ability to create the bounding box for only selected features
- the vector alg's exact extent calculation - we'd be using the (sometimes inaccurate, but faster) provider extent for the layer

**Comment by nirvn** [comment]
OK, thanks for the explanation, sounds logical.

File: python/plugins/processing/algs/help/qgis.yaml [hunk 1]
```diff
@@ -402,12 +402,14 @@ qgis:polygoncentroids: >
 
   NOTE: This algorithm is deprecated and the generic "centroids" algorithm (which works for line and multi geometry layers) should be used instead.
 
-
 qgis:polygonfromlayerextent: >
```
Comment (Gustry) [inline python/plugins/processing/algs/help/qgis.yaml]: Why not renaming this one as `polygonfromvectorextent` to be consistent with the new one?

Comment (nyalldawson) [reply python/plugins/processing/algs/help/qgis.yaml]: The id isn't shown anywhere - I've kept them as their 2.x versions just to (very slightly) minimise porting efforts for plugin authors

**Comment by nirvn** [comment]
@nyalldawson , just noticed a tiny inconsistency UI-wise, whereas polygon from {raster,vector} extent algs aren't in a similarly-named group (raster tools vs. vector general):
screenshot from 2017-08-31 09-37-09

**Pull request merged**